File: Course Handicap Calculator/Views/LabeledTextFieldView.m
#import "LabeledTextFieldView.h"


@implementation LabeledTextFieldView

- (instancetype)initWithCoder:(NSCoder *)aDecoder {
    self = [super initWithCoder:aDecoder];
    if (self) {
        [self prepareContentView];
    }
    return self;
}

- (instancetype)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    if (self) {
        [self prepareContentView];
    }
    return self;
}

- (void)prepareContentView {
    UINib *nib = [UINib nibWithNibName:NSStringFromClass([self class]) bundle:nil];
    UIView *contentView = [nib instantiateWithOwner:self options:nil].firstObject;
    
    contentView.frame = self.bounds;
    contentView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    
    [self addSubview:contentView];
}

@end

File: Course Handicap Calculator/Views/LabeledTextFieldView.swift
import UIKit


class LabeledTextFieldView: UIView {

    @IBOutlet var label: UILabel!
    @IBOutlet var textField: UITextField!

    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        self.prepareContentView()
    }
    
    override init(frame: CGRect) {
        super.init(frame:frame)
        self.prepareContentView()
    }

    private func prepareContentView() {
        let nib = UINib.init(nibName:"LabeledTextFieldView", bundle:nil)
        let contentView = nib.instantiate(withOwner:self, options:nil).first as! UIView
        contentView.frame = self.bounds
        contentView.autoresizingMask = [UIViewAutoresizing.flexibleWidth, UIViewAutoresizing.flexibleHeight]
        self.addSubview(contentView)
    }

}
